src/wui/soldierlist.cc

```diff
@@ -425,11 +425,11 @@ bool SoldierPanel::handle_mousemove(
 bool SoldierPanel::handle_mousepress(uint8_t btn, int32_t x, int32_t y) {
 	if (btn == SDL_BUTTON_LEFT) {
 		if (click_fn_) {
-			if (const Widelands::Soldier* soldier = find_soldier(x, y)) {
+			if (const Widelands::Soldier* soldier = find_soldier(x, y); soldier != nullptr) {
 				click_fn_(soldier);
+				return true;
 			}
 		}
-		return true;
```

Comment:
Not important, but I think it could still consume all left-clicks, couldn't it?

Reply:
Very important actually ;) This change is the fix for
   > Clicking and dragging on an empty part of the soldier list drags the window now.

Reply:
Indeed. Sorry.